src/main/java/frc/robot/subsystems/arm/ArmSystemModel.java

```diff
@@ -0,0 +1,161 @@
+        Kb_MATRIX.set(1, 0, 0);
+        Kb_MATRIX.set(0, 1, 0);
+    }
+
+    /**
+     * Calculates the M matrix.
+     *
+     * @param theta2 the angle of the elbow joint. [rad]
+     */
+    public void updateMMatrix(double theta2) {
+        double c2 = Math.cos(theta2);
+        double diagonal = m2 * (r2 * r2 + l1 * r2 * c2) + I2;
+
+        M_MATRIX.set(0, 0, m1 * r1 * r1 + m2 * (l1 * l1 + r2 * r2 + 2 * l1 * r2 * c2) + I1 + I2);
+        M_MATRIX.set(0, 1, diagonal);
+        M_MATRIX.set(1, 0, diagonal);
+        M_MATRIX.set(1, 1, m2 * r2 * r2 + I2);
+    }
+
+    /**
+     * Calculates the C matrix.
+     *
+     * @param theta2 the angle of the elbow joint. [rad]
+     * @param theta1Dot the angular velocity of the shoulder joint. [rad/s]
+     * @param theta2Dot the angular velocity of the elbow joint. [rad/s]
+     */
+    public void updateCMatrix(double theta2, double theta1Dot, double theta2Dot) {
+        double s2 = Math.sin(theta2);
+
+        C_MATRIX.set(0, 0, -m2 * l1 * r2 * s2 * theta2Dot);
+        C_MATRIX.set(0, 1, -m2 * l1 * r2 * s2 * (theta1Dot + theta2Dot));
+        C_MATRIX.set(1, 0, m2 * l1 * r2 * s2 * theta1Dot);
+        C_MATRIX.set(1, 1, 0);
+    }
+
+    /**
+     * Calculates the Tg vector.
+     *
+     * @param theta1 the angle of the shoulder joint. [rad]
+     * @param theta2 the angle of the elbow joint. [rad]
+     */
+    public void updateTgVector(double theta1, double theta2) {
+        double c1 = Math.cos(theta1);
+        double c12 = Math.cos(theta1 + theta2);
+
+        Tg_VECTOR.set(0, 0, (m1 * r1 + m2 * l1) * g1 * c1 + m2 * r2 * g2 * c12);
+        Tg_VECTOR.set(1, 0, m2 * r2 * g2 * c12);
+    }
+
+    /**
+     * Calculates the voltage feedforward required to move the arm in a certain state.
+     *
+     * @param theta1 the angle of the shoulder joint. [rad]
+     * @param theta2 the angle of the elbow joint. [rad]
+     * @param theta1Dot the angular velocity of the shoulder joint. [rad/s]
+     * @param theta2Dot the angular velocity of the elbow joint. [rad/s]
+     * @param theta1DotDot the angular acceleration of the shoulder joint. [rad/s^2]
+     * @param theta2DotDot the angular acceleration of the elbow joint. [rad/s^2]
+     * @return the voltage feedforward required to move the arm in the given state. [V]
+     */
+    public ArmFeedForward calculateFeedForward(double theta1, double theta2, double theta1Dot, double theta2Dot, double theta1DotDot, double theta2DotDot) {
+        updateMMatrix(theta2);
+        updateCMatrix(theta2, theta1Dot, theta2Dot);
+        updateTgVector(theta1, theta2);
+
+        var thetaDotVector = new SimpleMatrix(2, 1, MatrixType.DDRM);
+        thetaDotVector.set(0, 0, theta1Dot);
+        thetaDotVector.set(1, 0, theta2Dot);
+
+        var thetaDotDotVector = new SimpleMatrix(2, 1, MatrixType.DDRM);
+        thetaDotDotVector.set(0, 0, theta1DotDot);
+        thetaDotDotVector.set(1, 0, theta2DotDot);
```

Comment:
Perhaps create `thetaDotVector` using the `.copy()` method.

Reply:
What would it be copying? There aren't any variables that are identical to thetaDotVector

Reply:
@barel13, my intention was to let @Emma03L read my comment, then she could resolve the conversation after she would have read it.